src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -920,13 +919,13 @@ private Task WriteSuffix()
             // automatically for HEAD requests or 204, 205, 304 responses.
             if (_canHaveBody)
             {
-                if (!hasTransferEncoding && !responseHeaders.HasContentLength)
+                if (!hasTransferEncoding && !responseHeaders.ContentLength.HasValue)
                 {
                     if (appCompleted && StatusCode != StatusCodes.Status101SwitchingProtocols)
                     {
                         // Since the app has completed and we are only now generating
                         // the headers we can safely set the Content-Length to 0.
-                        responseHeaders.SetRawContentLength("0", _bytesContentLengthZero);
+                        responseHeaders.ContentLength = 0;
```

Comment:
Is an extra 8 byte field in every header set to make the occasional 1 byte write of `0` slightly faster (as most responses will have some content) - can always special case 0 in the copyfrom..?